### PR TITLE
[MM-16667] Migrate User.GetByUsername to Sync by default #11453

### LINKED ISSUE #11453: Migrate "User.GetByUsername" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-16667)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `GetByUsername` in the `User` store.

The expected way to implement it is, go to the `User` store implementation in the `store/sqlstore/` directory, modify the method `GetByUsername` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by Inconnu08**
#### Summary
Migrate `User.GetByUsername` to sync by default.

#### Ticket Link
Fixes #11453

### CODE REVIEW

**Comment by Inconnu08** [comment]
Thanks, guys!

File: store/storetest/user_store.go [hunk 1]
```diff
@@ -1671,33 +1671,33 @@ func testUserStoreGetByUsername(t *testing.T, ss store.Store) {
...
+		assert.Equal(t, u3, result)
 	})
 
 	t.Run("get by empty username", func(t *testing.T) {
-		result := <-ss.User().GetByUsername("")
-		require.NotNil(t, result.Err)
-		require.Equal(t, result.Err.Id, "store.sql_user.get_by_username.app_error")
+		result, err := ss.User().GetByUsername("")
+		require.NotNil(t, err)
+		require.Equal(t, result.Id, "store.sql_user.get_by_username.app_error")
```
Comment (jespino) [inline store/storetest/user_store.go]: Here is expected to check the `err` not the `result` you are having a null pointer there.

Comment (jespino) [reply store/storetest/user_store.go]: You are doing it all the way around. You should check the `err` not the `result`

Comment (Inconnu08) [reply store/storetest/user_store.go]: My apologies

**Comment by Inconnu08** [comment]
@jespino fixed

**Comment by jespino** [comment]
@Inconnu08 Thanks, now should be working

**Comment by Inconnu08** [comment]
Awesome! Will finish the next one's on my list shortly after.

**Pull request merged**